File: netease/deleteStr.py
def deleteStr(input):
    obj = {}
    strArr = list(input)
    for i in xrange(0, len(strArr)):
        outterArr = strArr[:]
        outterArr.pop(i)
        for j in xrange(0, len(outterArr)):
            innerArr = outterArr[:]
            innerArr.pop(j)
            currentStr = ''.join(innerArr)
            if not(currentStr in obj):
                obj[currentStr] = True
    return len(obj.keys())

print deleteStr('data')

File: netease/deleteStr.js
const deleteStr = (input)=>{
  let obj = {};
  let strArr = input.split('');
  for (let i = 0; i < strArr.length; i++) {
    let outterArr = strArr.slice(0);
    outterArr.splice(i,1);
    // console.log("outterArr",outterArr);
    for (let j = 0; j < outterArr.length; j++) {
      let innerArr = outterArr.slice(0);
      innerArr.splice(j,1);
      let currentStr = innerArr.join('');
      if (!obj[currentStr]) {
        obj[currentStr] = true;
      }
    }
  }
  return Object.keys(obj).length; 
}

console.log(deleteStr('data'));
console.log(deleteStr('abcdefgh'));
// deleteStr('data');